### ISSUE TITLE
Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column

### ISSUE THREAD

**Issue by loosi** [influxdata/telegraf#4312]
### Relevant telegraf.conf:
almost default mysql.conf
```
[[inputs.mysql]]
  servers = ["USER:PW@tcp(127.0.0.1:3306)/?tls=false"]
  metric_version = 2
  perf_events_statements_digest_text_limit  = 120
  perf_events_statements_limit              = 250
  perf_events_statements_time_limit         = 86400
  table_schema_databases                    = []
  gather_table_schema                       = true
  gather_process_list                       = true
  gather_user_statistics                    = true
  gather_info_schema_auto_inc               = false
  gather_innodb_metrics                     = true
  gather_slave_status                       = false
  gather_binary_logs                        = false
  gather_table_io_waits                     = true
  gather_table_lock_waits                   = true
  gather_index_io_waits                     = false
  gather_event_waits                        = false
  gather_file_events_stats                  = false
  gather_perf_events_statements             = false
  interval_slow                             = "10m"
```

### System info:
* ubuntu 16.04
* telegraf 1.7.0-1 (influxdata repo)
* percona xtraDB 5.7.21 cluster with *only_full_group_by*

### Steps to reproduce:
run telegraf with mysql config enabled

### Expected behavior:
no error?

### Actual behavior:
throwing errors 
```
MySQL Error gathering user stats: Error 1055: Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'information_schema.USER_STATISTICS.TOTAL_CONNECTIONS' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
```

Well that error mostly explains itself, obviously the "GROUP BY" clause is incomplete. So add all columns into the "GROUP BY" clause. I will open a pull request for that

**Comment by glinton** [comment]
I think the easiest fix for this is to remove the `GROUP BY user` entirely as the `USER_STATISTICS` table already effectively groups by `user`. I'll incorporate that into #4306

**Comment by loosi** [comment]
Okay, I'm fine with that. I removed the PR

**Comment by glinton** [comment]
Resolved via #4306
